### PR TITLE
Add migration documents from ASM

### PR DESCRIPTION
**Pull request by tashirosota**
https://github.com/okuramasafumi/alba/issues/98

### CODE REVIEW

**Comment by tashirosota** [comment]
@okuramasafumi 
Hi. I made a template for the time being.
Let me check some,
```
 - Is the location correct?
 - How pattern do you want migration case?
 - Is there anything you would like to mention in particular?
```

**Comment by okuramasafumi** [comment]
> Is the location correct?

`doc` directory is fine, but this directory is also used by `yard` (that's why it's ignored).
We could use `docs` directory so that we can separate handwritten documents with autogenerated documents.

Also I want to mention that file name should be something like "migrate_from_jbuilder", not just "jbuilder".

> How pattern do you want migration case?

I think we need migration document for real-world, complex cases. For example, with jbuilder we use partials a lot so I'd like to mention how to write partial-like serializer with Alba.

> Is there anything you would like to mention in particular?

It's important to mention that Alba doesn't have all the features AMS or jbuilder have. For example, Alba doesn't support cache. I think it's fair to mention some feature Alba doesn't have.

**Comment by tashirosota** [comment]
@okuramasafumi 
Thanks for review！I'll fix in this week.

**Comment by tashirosota** [comment]
@okuramasafumi 
I'm sorry it will take longer than expected!
If I'm in a hurry,  I'd like to divide the PR for each serializer, but what about?
I will start with the ActiveModelSerializers that we use all the time.

**Comment by okuramasafumi** [comment]
@tashirosota Yeah just adding a document for AMS is fine.

**Comment by tashirosota** [comment]
@okuramasafumi 
Hi okura.
I usually use ASM, couldn't find unsupported ASM's features, if you know then, teach me.
And please review for pr.

File: docs/migrate_from_active_model_serializers.md [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+  attribute :email do
+    object.profile.email
+  end
+end
+
+# serialze
+CustomUserResource.new(user).serializable_hash
+=> {
+  email: email
+}
```
Comment (okuramasafumi) [inline docs/migrate_from_active_model_serializers.md:286]:
```suggestion
# => {
#   email: email
# }
```

File: docs/migrate_from_active_model_serializers.md [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+  :updated_at=>updated_at,
+  :custom_params=>{
+    :given_params=>{
+      :a=>:b
+    }
+  }
+}
+```
+
+#### When Alba
```
Comment (okuramasafumi) [inline docs/migrate_from_active_model_serializers.md:321]:
```suggestion
#### Alba
```

File: docs/migrate_from_active_model_serializers.md [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+=> {
+  :id=>1,
+  :created_at=>created_at,
+  :updated_at=>updated_at,
+  :custom_params=>{
+    :given_params=>{
+      :a=>:b
+    }
+  }
+}
```
Comment (okuramasafumi) [inline docs/migrate_from_active_model_serializers.md:318]:
```suggestion
# => {
#   :id=>1,
#   :created_at=>created_at,
#   :updated_at=>updated_at,
#   :custom_params=>{
#     :given_params=>{
#       :a=>:b
#     }
#   }
# }
```

File: docs/migrate_from_active_model_serializers.md [hunk 4]
```diff
@@ -0,0 +1,356 @@
...
+  params: { a: :b } # Give with :params keyword argument
+).serializable_hash
+=> {
+  :id=>1,
+  :created_at=>created_at,
+  :updated_at=>updated_at,
+  :custom_params=>{
+    :a=>:b
+  }
+}
```
Comment (okuramasafumi) [inline docs/migrate_from_active_model_serializers.md:347]:
```suggestion
# => {
#   :id=>1,
#   :created_at=>created_at,
#   :updated_at=>updated_at,
#   :custom_params=>{
#     :a=>:b
#   }
# }
```

**Comment by okuramasafumi** [comment]
@tashirosota I suggested multiple changes including fixing typos and commenting out outputs.

https://github.com/rails-api/active_model_serializers/blob/0-10-stable/docs/howto/serialize_poro.md
This document shows that serializing PORO with AMS is pretty hard. Should we use ActiveModel objects for examples instead?

File: docs/migrate_from_active_model_serializers.md [hunk 5]
```diff
@@ -0,0 +1,357 @@
...
+2. Complex serialization
+3. Unsupported features
+
+## Example class
+
+TODO:
+Example clsss is included `ActiveModel::Model`, because [serializing PORO with AMS is pretty hard](https://github.com/rails-api/active_model_serializers/blob/0-10-stable/docs/howto/serialize_poro.md).This example also works with ActiveRecord.
+
+```rb
+class User
```
Comment (tashirosota) [inline docs/migrate_from_active_model_serializers.md:19]: @okuramasafumi 
I can't find how to serialize with object included ActiveModel::Model.
https://github.com/rails-api/active_model_serializers/tree/0-10-stable#high-level-behavior
`SomeResource < ActiveModelSerializers::Model` is a bad example of both compatibility, so I'd like to use `SomeResource < ActiveRecord::Base`, but is that okay?
Like this.
```rb
class User < ActiveRecord::Base
  # columns: id, created_at, updated_at
  has_one :profile
  has_many :articles
end
```

**Review by okuramasafumi — APPROVED** [review]
Thank you!

File: docs/migrate_from_active_model_serializers.md (hunk 5 above)
Comment (okuramasafumi) [reply docs/migrate_from_active_model_serializers.md]: Sorry for the late reply.
Yes, using `AR::Base` is fine.

**Pull request merged**